### PR TITLE
VideoClickAction api

### PR DESCRIPTION
**Pull request by C10udburst**
TODO
- [x] move all open in app to new api
- [x] fix isCasting property in some places
- [x] fix default player setting
- [ ] move actions to extensions
- [x] move fcast to new api
- [x] clean unused strings

possible examples of new actions
- download with 1DM
- kdeconnect
- cast with airplay

### CODE REVIEW

File: app/src/main/java/com/lagradost/cloudstream3/actions/temp/MpvKtPackage.kt [hunk 1]
```diff
@@ -0,0 +1,69 @@
+package com.lagradost.cloudstream3.actions.temp
```
Comment (fire-light42) [inline app/src/main/java/com/lagradost/cloudstream3/actions/temp/MpvKtPackage.kt:1]: We have like 10 mpv forks, so just make a master class that these inherit from.

File: app/src/main/java/com/lagradost/cloudstream3/actions/OpenInAppAction.kt [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        if (intentClass != null) {
+            intent.component = ComponentName(packageName, intentClass)
+        }
+        putExtra(context, intent, video, result, index)
+        setKey("last_opened_id", video.id)
+        try {
+            CoroutineScope(Dispatchers.IO).launch {
+                activityResultLauncher?.launch(intent)
+            }
+        } catch (t: Throwable) {
```
Comment (fire-light42) [inline app/src/main/java/com/lagradost/cloudstream3/actions/OpenInAppAction.kt:115]: We can catch exceptions like this to clarify what is happening.

```kotlin
catch (_: ActivityNotFoundException) {
showToast(R.string.app_not_found_error, Toast.LENGTH_LONG)
} catch (t: Throwable) { 
logError(t)
showToast(t.toString(), Toast.LENGTH_LONG) 
}
```
`main {}` should also not be required as showToast is thread safe when you dont supply an activity. txt can also be discarded as showToast allows @ stringres.

**Review by fire-light42 — CHANGES_REQUESTED** [review]
Also remove player_pref_names and such from array.xml

Overall, a very nice PR, I may request more changes after testing.

File: app/src/main/java/com/lagradost/cloudstream3/actions/OpenInAppAction.kt (hunk 2 above)
Comment (fire-light42) [reply app/src/main/java/com/lagradost/cloudstream3/actions/OpenInAppAction.kt:115]: Resolve without commit?

Comment (C10udburst) [reply app/src/main/java/com/lagradost/cloudstream3/actions/OpenInAppAction.kt:115]: sorry, i will commit soon. i didn't know resolving in android studio will resolve it remotely

File: app/src/main/java/com/lagradost/cloudstream3/actions/temp/fcast/FcastAction.kt [hunk 3]
```diff
@@ -13,7 +14,7 @@ import com.lagradost.cloudstream3.utils.ExtractorLinkType
 import com.lagradost.cloudstream3.utils.SingleSelectionHelper.showBottomDialog
 
 class FcastAction: VideoClickAction() {
-    override val name = txt("Fcast")
+    override val name = txt("Fcast to device")
```
Comment (fire-light42) [inline app/src/main/java/com/lagradost/cloudstream3/actions/temp/fcast/FcastAction.kt:17]: Translation?

File: app/src/main/java/com/lagradost/cloudstream3/actions/temp/PlayInBrowserAction.kt [hunk 4]
```diff
@@ -11,7 +12,7 @@ import com.lagradost.cloudstream3.ui.result.txt
 import com.lagradost.cloudstream3.utils.ExtractorLinkType
 
 class PlayInBrowserAction: VideoClickAction() {
-    override val name = txt("Play in browser")
+    override val name = txt(R.string.episode_action_play_in_format, "Browser")
```
Comment (fire-light42) [inline app/src/main/java/com/lagradost/cloudstream3/actions/temp/PlayInBrowserAction.kt:15]: Translation?

File: app/src/main/java/com/lagradost/cloudstream3/actions/temp/ViewM3U8Action.kt [hunk 5]
```diff
@@ -2,14 +2,17 @@ package com.lagradost.cloudstream3.actions.temp
...
 import com.lagradost.cloudstream3.actions.VideoClickAction
 import com.lagradost.cloudstream3.actions.makeTempM3U8Intent
 import com.lagradost.cloudstream3.ui.result.LinkLoadingResult
 import com.lagradost.cloudstream3.ui.result.ResultEpisode
 import com.lagradost.cloudstream3.ui.result.txt
 
-class TestAction: VideoClickAction() {
-    override val name = txt("Test action")
+class ViewM3U8Action: VideoClickAction() {
+    override val name = txt(R.string.episode_action_play_in_format, "m3u8 player")
```
Comment (fire-light42) [inline app/src/main/java/com/lagradost/cloudstream3/actions/temp/ViewM3U8Action.kt:13]: Translation?

File: app/src/main/java/com/lagradost/cloudstream3/actions/temp/CopyClipboardAction.kt [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+import android.content.Context
+import com.lagradost.cloudstream3.actions.VideoClickAction
+import com.lagradost.cloudstream3.ui.result.LinkLoadingResult
+import com.lagradost.cloudstream3.ui.result.ResultEpisode
+import com.lagradost.cloudstream3.ui.result.txt
+import com.lagradost.cloudstream3.utils.UIHelper.clipboardHelper
+
+class CopyClipboardAction: VideoClickAction() {
+    override val name = txt("Copy to clipboard")
```
Comment (fire-light42) [inline app/src/main/java/com/lagradost/cloudstream3/actions/temp/CopyClipboardAction.kt:11]: Translation?

**Review by fire-light42 — COMMENTED** [review]
Code looks good, but dont hardcode strings that cant be translated

**Comment by C10udburst** [comment]
I will implement translations after transitioning to extension, since resource loading is a bit different there. Before I move the actions there the API itself needs to be in prerelease.

**Comment by fire-light42** [comment]
Please don't merge, before I test and review it. This PR now crashes the prerelease version due to unhandled exceptions. I will make a commit to fix this soon.

**Pull request merged**